commit d7fb120774f062ce7db439863ab5d4190d6f989c
Author: Trond Myklebust <Trond.Myklebust@netapp.com>
Date:   Mon Oct 6 20:08:56 2008 -0400

    NFS: Don't use range_cyclic for data integrity syncs
    
    It is more efficient to write linearly starting from the beginning of the
    file.
    
    Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>

diff --git a/fs/nfs/write.c b/fs/nfs/write.c
index 3229e21..9f98458 100644
--- a/fs/nfs/write.c
+++ b/fs/nfs/write.c
@@ -1427,8 +1427,9 @@ static int nfs_write_mapping(struct address_space *mapping, int how)
 		.bdi = mapping->backing_dev_info,
 		.sync_mode = WB_SYNC_NONE,
 		.nr_to_write = LONG_MAX,
+		.range_start = 0,
+		.range_end = LLONG_MAX,
 		.for_writepages = 1,
-		.range_cyclic = 1,
 	};
 	int ret;
 
